txpool/txpool_test.go

```diff
@@ -121,3 +122,100 @@ func TestTxnQueue_Promotion(t *testing.T) {
 	assert.Equal(t, nonce, uint64(2))
 	assert.Equal(t, pool.Length(), uint64(2))
 }
+
+func TestTxnQueue_Heap(t *testing.T) {
+	type TestCase struct {
+		From     types.Address
+		GasPrice *big.Int
+		Nonce    uint64
+		Index    int
+	}
+
+	addr1 := types.Address{0x1}
+	addr2 := types.Address{0x2}
+
+	test := func(t *testing.T, testTable []TestCase) {
+		pool, err := NewTxPool(hclog.NewNullLogger(), false, &mockStore{}, nil, nil)
+		assert.NoError(t, err)
+		pool.EnableDev()
+
+		for _, testCase := range testTable {
+			err := pool.addImpl("", &types.Transaction{
```

Comment:
Why didn't you test the public method `AddTx` instead of the private one?

Reply:
The rest of the tests use the internal `addImpl`, because `AddTx` is just sugar coat:
- it calls `addImpl` internally
- it broadcasts the transaction
---
https://github.com/0xPolygon/polygon-sdk/blob/acfa430342569c1c74486e2b427a96e9ab908f02/txpool/txpool.go#L123-L148

Since these are unit tests and we don't have networking or broadcasts enabled for this specific test, calling `AddTx` is basically the same as calling `addImpl`.

Reply:
It was only a comment in the sense that it's more important to test public than private methods. I understand that currently it is the same, but tomorrow if someone is to change the "wrapper" method (`AddTx`) we wouldn't catch that.

No need to focus on this right now, just to keep it in mind for future testing - testing public methods should take precedence over private ones since the client (or the rest of the system) will use these, and not private ones 🙂

Reply:
I completely agree. We need to overhaul and standardize the way we write / run most tests in the SDK. 
We'll probably open up several tasks further down the line to refactor these tests :+1: